fix build when job control is disabled

http://lists.gnu.org/archive/html/bug-bash/2011-06/msg00110.html
http://lists.gnu.org/archive/html/bug-bash/2011-07/msg00057.html
http://lists.gnu.org/archive/html/bug-bash/2011-09/msg00039.html
https://bugs.gentoo.org/383237

should be fixed in next release of bash ...

--- a/execute_cmd.c
+++ b/execute_cmd.c
@@ -2202,7 +2202,11 @@
   /* If the `lastpipe' option is set with shopt, and job control is not
      enabled, execute the last element of non-async pipelines in the
      current shell environment. */
-  if (lastpipe_opt && job_control == 0 && asynchronous == 0 && pipe_out == NO_PIPE && prev > 0)
+  if (lastpipe_opt && 
+#if defined(JOB_CONTROL)
+		job_control == 0 && 
+#endif
+		asynchronous == 0 && pipe_out == NO_PIPE && prev > 0)
     {
       lstdin = move_to_high_fd (0, 0, 255);
       if (lstdin > 0)
